### PR TITLE
[DPE-2610] Remove support of Juju 2.9

### PR DESCRIPTION
**Pull request by dmitry-ratushnyy**
The goal of this PR is to remove support of Juju 2.9 from the charm.

To do so we are removing code related to handling secrets for juju 2.9

### CODE REVIEW

**Review by MiaAltieri — CHANGES_REQUESTED** [review]
looks good, but dont forget to add
```
          juju-channel: 3.1/stable
          bootstrap-options: "--agent-version 3.1.5"
```
to ci or the tests will fail. [you can use vm CI as a reference](https://github.com/canonical/mongodb-operator/blob/6/edge/.github/workflows/ci.yaml)

File: pyproject.toml [hunk 1]
```diff
@@ -45,7 +45,7 @@ lightkube = "^0.13.0"
 pytest = "^7.3.1"
 pytest-mock = "^3.11.1"
 pytest-operator = "^0.27.0"
-juju = "2.9.44.0 || 3.2.0.1"
+juju = "3.2.0.1"
```
Comment (delgod) [inline pyproject.toml:48]: maybe, can be separate PR
```suggestion
juju = "3.2.2"
```

File: tox.ini [hunk 2]
```diff
@@ -70,7 +70,7 @@ pass_env =
     CI
 commands_pre =
     poetry install --with integration
-    poetry run pip install juju==2.9.44.0
+    poetry run pip install juju==3.2.0.1
```
Comment (delgod) [inline tox.ini:73]: same as above
```suggestion
    poetry run pip install juju==3.2.2
```

**Review by delgod — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (dmitry-ratushnyy) [reply pyproject.toml:48]: Will commit in this one  if you don't mind

**Review by MiaAltieri — DISMISSED** [review]

**Review by delgod — APPROVED** [review]
